### ISSUE TITLE
Linux aarch64 support

### ISSUE THREAD

**Issue by ajm-ska** [CARTAvis/carta-backend#850]
I was the proponent for ARM support before, and I thought having Apple Silicon support would be enough. At the time we added the following to the CMakeLists.txt file:

> if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm" AND ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
>     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_ARM_ARCH_ -march=armv8-a+fp+simd+crypto+crc")

With those lines, the carta_backend builds and runs fine on an Apple Silicon Mac because cmake reads `CMAKE_SYSTEM_PROCESSOR` as `arm` and `CMAKE_SYSTEM_NAME` as `Darwin`.

At the time, I didn't think it would be necessary to support ARM on Linux. The only Linux based ARM system I tried was a Raspberry Pi and although carta_backend can build and run on it, the compile time is extremely slow, so I thought is was not very practical for developing or running CARTA.

However, now Docker is officially available on Apple Silicon. It could easily be used to develop or run CARTA through a Linux Docker container on an Apple Silicon Mac.

In Docker on Apple Silicon, cmake reads `CMAKE_SYSTEM_PROCESSOR` as `aarch64` and `CMAKE_SYSTEM_NAME` as `Linux`. Therefore, I propose to add the following addtional elseif line to the CMakeLists.txt:

> elseif(${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64" AND ${CMAKE_SYSTEM_NAME} MATCHES "Linux")
>     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_ARM_ARCH_ -march=armv8-a+fp+simd+crypto+crc")

The only thing I am unsure of are the optimisation flags to use. The same flags that we use natively on Apple Silicon MacOS appear to work fine in the Docker Linux container, and they also appear to work fine on the Raspberry Pi. 
Perhaps we could leave the optimisation flags as they are now, and consider adjusting them in future when there is more information or more time to benchmark performance?

In summary, I think it would be very useful if the carta_backend can simply compile and run on any combination of x86-64/arm/aarch64/Darwin/Linux system without the user needing to modify the CMakeLists.txt beforehand. 
It might also allow the Ubuntu Launchpad to create aarch64 Debian packages?
